core/controllers/learner_playlist.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the learner playlist."""
+
+from core.controllers import base
+from core.domain import learner_progress_services
+from core.domain import learner_playlist_services
+
+
+class AddExplorationToLearnerPlaylistHandler(base.BaseHandler):
```

Comment:
I think this is too many handlers. Can we collapse into two: AddActivityToLearnerPlaylistHandler and RemoveActivityFromLearnerPlaylistHandler? Pass in activity type and activity ID from the frontend.

EDIT: see also later comments in main.py.